src/Intake.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Intake.hpp"
+
+#include "constants.hpp"
+
+Intake::Intake(uint8_t const left_port, uint8_t const right_port) : m_left_motor{ left_port }, m_right_motor{ right_port, true } {}
+
+bool Intake::on() const {
+	return m_on;
+}
+
+void Intake::set_on(bool value) {
+	m_on = value;
+	auto velocity = 0;
+	if (m_on) {
+		velocity = constants::INTAKE_VELOCITY;
+	}
+	m_left_motor.move_velocity(velocity);
```

Comment:
The way the code is implemented, this would only move on button press. Is this intended behavior?

Reply:
`move_velocity` runs in the background. Every time the button is pressed the motor velocity is toggled, which stays in effect until the button is pressed again. This is intended behavior.

Reply:
Oh yeah, I forgot that lol. Everything looks good to me!